lib/saml_idp/xml_security.rb

```diff
@@ -179,56 +176,26 @@ def validate_doc_embedded_signature(
       )
         # check for inclusive namespaces
         inclusive_namespaces = extract_inclusive_namespaces
-        document = Nokogiri.parse(to_s)
 
-        # create a working copy so we don't modify the original
-        @working_copy ||= REXML::Document.new(to_s).root
 
-        # store and remove signature node
-        @sig_element ||= begin
-          element = REXML::XPath.first(
-            @working_copy,
-            '//ds:Signature',
-            { 'ds' => DSIG }
-          )
-          element.remove
-        end
+        sig_element = document.at_xpath('//*:Signature')
+        sig_namespace_hash = sig_element.at_xpath('//*:SignedInfo')&.namespaces
+        noko_signed_info_element = sig_element.at_xpath('./*:SignedInfo')
```

Comment:
Minor: Remove noko_ prefix?

Reply:
oh yeah great catch, thanks

Reply:
updated in https://github.com/18F/saml_idp/pull/110/commits/54d2d1565f7ce201ca494272df69bd6ab4b9cc5f